python/lib/dependabot/python/language_version_manager.rb

```diff
@@ -33,8 +42,12 @@ def install_required_python
       end
 
       def python_major_minor
-        @python ||= Python::Version.new(python_version)
-        "#{@python.segments[0]}.#{@python.segments[1]}"
+        @python ||= extract_python_major_minor(python_version)
+      end
+
+      def self.extract_python_major_minor(python_version_string)
+        full_version = Python::Version.new(python_version_string)
+        "#{@full_version.segments[0]}.#{@full_version.segments[1]}"
```

Comment:
I extracted this to a class method as a helper as there are multiple places we could use this... but Rubocop is complaining about it. I'm not fully understanding why... or rather, the bits that I do understand after reading up on it leave me a little frustrated...

This may require refactoring first such that `@python` is memoized separately outside of this method. 🤔

Reply:
https://github.com/dependabot/dependabot-core/pull/7703